custom_components/trakt_tv/apis/trakt.py

```diff
@@ -83,37 +119,55 @@ async def fetch_upcoming(self, trakt_kind: TraktKind, all_medias: bool):
         path = trakt_kind.value.path
         identifier = trakt_kind.value.identifier
 
-        if not configuration.upcoming_identifier_exists(identifier, all_medias):
+        if (not configuration.upcoming_identifier_exists(identifier, all_medias)) and not (configuration.next_to_watch_identifier_exists(identifier)):
             return None
 
         configuration = Configuration(data=self.hass.data)
-        days_to_fetch = configuration.get_upcoming_days_to_fetch(identifier, all_medias)
+
+        if not next_to_watch:
```

Comment:
prefere starting with `if next_to_watch` then in the else you can do the rest better for the brain ^^